Cli refactor

mike0sv opened the pull request:
closes #153 
closes #100 

Checklist:
**Common**:
- [x] use rich for better formatting
- [x] stream errors (traceback) to stderr if --verbose is set
- [x] url to issues page in case of unexpected error
- [x] color/bold for important stuff
- [x] emojis at the beginning of line for important stuff (but no overuse)
- [x] Communicate the result of operations where applicable
- [ ] assign codes to different exceptions and use them as exit codes
- [ ] probably some additional code for better autocompletion
- [x] config to disable emoji

**apply**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**clone**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**config set**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**create**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**deploy**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] move to deploy subcommand
- [ ] emojis

**destroy**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] move to deploy subcommand
- [ ] emojis

**status**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [ ] --json
- [x] move to deploy subcommand
- [ ] emojis

**import**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**ls**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] --json
- [ ] emojis

**pprint**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] --json
- [x] emojis

**init**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] ascii logo at init
- [x] emojis

**link**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

**pack**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [ ] emojis

**serve**

- [x] help
- [x] args help
- [x] tests
- [x] examples
- [x] emojis

mike0sv committed:
migration to typer

mike0sv committed:
typer tests

mike0sv committed:
update black

mike0sv committed:
bump all of pre-commit

mike0sv committed:
pin click

mike0sv committed:
fix test and nice init message

mike0sv committed:
add rich and work on apply

mike0sv committed:
add echo func and apply help/output

mike0sv committed:
examples in help
clone help and tests

mike0sv committed:
remote config and config cli
closes #116

mike0sv committed:
types cli and create cli help

mike0sv committed:
command sections and all the helps and examples

mike0sv committed:
command sections and all the helps and examples

mike0sv committed:
fix tests

mike0sv committed:
fix tests

mike0sv committed:
fix tests

mike0sv committed:
fix windows bugs

mike0sv committed:
cli deploy tests

mike0sv committed:
test for init, pack, serve

mike0sv committed:
fix windows bugs

mike0sv committed:
json options for ls and print

aguschin reviewed (commented):
Great work! Looks even more awesome now 😂 
Few more notes:
1. can we make `mlem -h` output help?
2. `dvc ls` is an alias for `dvc list`. Should we add it as alias also?
3. Try `mlem list dsadas` - it outputs full traceback.

mike0sv committed:
fix tests

mike0sv committed:
optimize ls a little
add aliases for cmd
better help for cli app
better cli exception handling
clone bugs fixed

mike0sv committed:
artifacts refactor

mike0sv committed:
fix tests

mike0sv committed:
fix tests

mike0sv committed:
fix tests and examples

mike0sv committed:
fix tests pylint

mike0sv committed:
clone target option to argument

mike0sv committed:
allow clone target to be "." or empty
ls in local fs will find repo
--json for apply
Choices api
lowercase metavars

mike0sv committed:
fix ls

mike0sv committed:
fix pylint

mike0sv committed:
add tests for help
